docs/EVALUATION.md

```diff
@@ -0,0 +1,249 @@
+        model 5          :a5, after a4, 25y
+        Trigger6 :milestone, :after a5
+        model 6          :a6, after a5, 5y
+        Trigger7 :milestone, :after a6
+        model 7          :a7, after a6, 5y
+    section evaluations
+        Evaluation1 :milestone, 1940, 0d
+        Evaluation2 :milestone, 1945, 0d
+        Evaluation3 :milestone, 1950, 0d
+        Evaluation4 :milestone, 1975, 0d
+        Evaluation5 :milestone, 2000, 0d
+        Evaluation6 :milestone, 2005, 0d
+        Evaluation7 :milestone, 2010, 0d
+    section evaluation data
+        data1 :e1, 1930, 10y
+        data2 :e2, after e1, 5y
+        data3 :e3, after e2, 5y
+        data4 :e4, after e3, 25y
+        data5 :e5, after e4, 25y
+        data6 :e6, after e5, 5y
+        data7 :e7, after e6, 5y
+```
+
+##### Proposed solution
+
+We add a new evaluation strategy with an according pipeline stage that runs evaluation after the rest of the core pipeline finished. By doing so we have access to all trigger times and can evaluate the models in the correct time ranges. After running evaluations on every "triggering-interval" (time between two triggers), we can calculate the pipeline performance as the weighted average of the model performances accounting for the number of samples in the time range.
+
+> Hypothesis: There's no real need to run evaluations directly after the training. We can always do that after the core pipeline finished. If required we can also make use of information like `first_timestamp` and `last_timestamp` from the logs for a specific trigger.
+
+#### [1] Model performance at a certain point in time
+
+We assume the performance of a model changes continuously over time and not at discrete points in time (like the trigger times).
+
+As we want to be able to plot the performance (i.e. accuracy) over the time range of the pipeline run, we have to observe the performance at a higher precision than [2] demands.
+
+Plotting with evenly spaced triggering indexes as x-axis (worst case is barplot where everything between two triggers is aggregated) is not sufficient as it disregards performance change in the potentially long intervals between triggers.
+
+Otherwise we could only plot a horizontal bar for every model evaluation at a trigger time. This would not allow us to see how the performance changes over time within one triggering interval. This information is important to fairly evaluate a trigger policy though as it could happen that a severe data drift causes rapid performance degradation. If the model wouldn't trigger for some time though we want to see that in the plot! Only plotting the total performance over the time range would hide this information.
+
+We therefore want to evaluate the model at multiple points in time within the triggering interval. We could define a configuration option that would allow us to either set the number of evaluation steps between two triggers or the time interval or number of samples between two evaluations.
```

Comment:
Can't we just timeslice the evaluation dataset differently? 

Although I see that in yearbook e.g. the timestamps themselves are on a yearly basis. Then again, I wonder what is the benefit of finer grained accuracy evaluation since for all benchmark datasets I am aware of, we have the timestamps on a certain granularity, and our evaluation already runs on that granularity (e.g., years). I am not sure about CLOC (there we could have finegrained timestamps, need to check).  Hence, I am not sure if you want to spend time on this (I might be missing something here though). If we'd have very finegrained timestamps maybe... it really depends on what you focus on I think. As I mention above, what I like about the yearly aggregation is that we have a common reference point for different triggering policies: For each year, given whatever is the most recent model, what is its performance on that timeframe? That allows us to compare all policies on a shared x axis. I am not sure how that would be possible with an non-evenly spaced x-axis.